### PR TITLE
DEV: Add `pluginAPI` method for adding buttons to table wrapper

### PR DESCRIPTION
**Pull request by keegangeorge**
> ⚠️ WIP
This PR adds the ability to add buttons to a post's table wrapper easily using a new `pluginAPI` method.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/lib/plugin-api.js [hunk 1]
```diff
@@ -2014,6 +2018,31 @@ class PluginApi {
...
+   *
+   *  api.addTableWrapperButton(
+   *    themePrefix("discourse_table_builder.edit.btn_edit"),
+   *    "pencil-alt",
+   *    ["btn-edit-table"],
+   *    generateModalEvent
+   * );
+   *
+   * @param {string} label - An i18n key for the button label
+   * @param {icon} icon - An icon name from fontawesome
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/lib/plugin-api.js:2036]: Should be string no? We don't have "icon" type

```suggestion
   * @param {string} icon - An icon name from fontawesome
```

Comment (keegangeorge) [reply app/assets/javascripts/discourse/app/lib/plugin-api.js:2036]: A yes good catch, thanks!

**Comment by jjaffeux** [comment]
@keegangeorge is it still something you want to work on?

**Comment by keegangeorge** [comment]
> @keegangeorge is it still something you want to work on?

I haven't been able to get this to work correctly for multiple tables and I'm not sure it's even possible. I'm going to close this out for now and perhaps revisit this later.

**Pull request closed without merging**